test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs

```diff
@@ -0,0 +1,488 @@
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var link = new ODataEntityReferenceLink() { Url = new Uri(_baseUri + "Orders(-10)") };
+                await messageWriter.WriteEntityReferenceLinkAsync(link);
+
+                // No exception is expected. Simply verify the writing succeeded.
+                if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                {
+                    Stream stream = await responseMessageWithoutModel.GetStreamAsync();
+                    Assert.Contains("$metadata#$ref", await TestsHelper.ReadStreamContentAsync(stream));
+                }
+            }
+
+            // write reference links without serialization info
+            responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var links = new ODataEntityReferenceLinks()
+                {
+                    Links = new[]
+                    {
+                        new ODataEntityReferenceLink(){Url = new Uri(_baseUri + "Orders(-10)")}
+                    },
+                };
+
+                await messageWriter.WriteEntityReferenceLinksAsync(links);
+
+                // No exception is expected. Simply verify the writing succeeded.
+                if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                {
+                    Stream stream = await responseMessageWithoutModel.GetStreamAsync();
+                    Assert.Contains("$metadata#Collection($ref)", (await TestsHelper.ReadStreamContentAsync(stream)));
+                }
+            }
+
+            // write request message containing an entry
+            var requestMessageWithoutModel = new TestStreamRequestMessage(new MemoryStream(),new Uri(_baseUri + "People"), "POST");
+            requestMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            using (var messageWriter = new ODataMessageWriter(requestMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceWriterAsync();
+                var entry = this.CreatePersonEntryWithoutSerializationInfo();
+                await odataWriter.WriteStartAsync(entry);
+                await odataWriter.WriteEndAsync();
+                Stream stream = await requestMessageWithoutModel.GetStreamAsync();
+                Assert.Contains("People(-5)\",\"PersonId\":-5,\"Name\":\"xhsdckkeqzvlnprheujeycqrglfehtdocildrequohlffazfgtvmddyqsaxrojqxrsckohrakdxlrghgmzqnyruzu\"", await TestsHelper.ReadStreamContentAsync(stream));
+            }
+        }
+
+        /// <summary>
+        /// Specify serialization info for both feed and entry should fail.

```

Comment:
What exactly is this test testing? It says that specify serialization for both feed and entry should fail, however, there are no failure assertion in the test. I'm not sure I understand the exact behaviour that we're evaluating here (or in this class in general).

Reply:
The test shows specifying serialization info for both a feed and entry should fail. I've updated the doc and test name.